AlexP-Elastic/dockerized-idp-testbed
# DO NOT MERGE - Show the differences needed to get working + integrated with ES/Kib

Related: https://github.com/elastic/cloud/issues/11401

To run this branch (eg in a VM):
* Add `idptestbed` to the `/etc/hosts` 
   * (eg as one of the internal IPs of the host - would need to change that in a few places to get it working as something externally addressible)
* In the top-level directory:
   * run `docker-compose build` (first time/after any changes to the config files) 
   * `docker-compose up` (subsequent times)
      * (note that I had to `docker compose rm` after `CTRL+C`ing)
* In the elasticsearch testbed directory:
   * `docker-compose up`
* Then bring up a browser on the VM and navigate to `http://idptestbed:5601` to login to Kibana (`https://idptestbed` will take you to the default IDP testbed)
   * Login as any username, with the same password
      * (eg `alex`/`alex` authenticates, `alex`/`banana` doesn't)
   * note that I didn't add any groups so this only shows authentication as `uid: 1`
      * You can change code in [`dummy-jaas-loginmodule`](https://github.com/AlexP-Elastic/dummy-jaas-loginmodule/blob/alexp-sample-login-module/src/main/java/net/unicon/authn/jaas/NotAsSimpleLoginModule.java#L55
) to add other attributes (which then need to get added to the attrbute resolver, [eg](https://github.com/AlexP-Elastic/dockerized-idp-testbed/pull/1/files#diff-276ed6f3df12b04e106e6bf5cf33cb3bR33), (`principalAttributeName` is the name it's been given in `dummy-jaas-loginmodule`)
